src/Formatter/CurlCommandFormatter.php

```diff
@@ -31,9 +31,13 @@ public function formatRequest(RequestInterface $request)
             $command .= sprintf(' -H \'%s: %s\'', $name, $request->getHeaderLine($name));
         }
 
-        $body = $request->getBody()->__toString();
-        if (!empty($body)) {
-            $command .= sprintf(' --data \'%s\'', escapeshellarg($body));
+        $body = $request->getBody();
+        if ($body->getSize() > 0) {
+            if (!$body->isSeekable()) {
+                throw new \RuntimeException('Cannot take data from a stream that is not seekable');
```

Comment:
Sending an exception here is horrible, i.e. some requests can randomly fail only on debugging (not on prod)  in a symfony env with this formatter in the debug toolbar. Can we set a specific exception like : CannotFormatException (or something else) that we can catch in class using this formatter to not show something ?